### ISSUE TITLE
It's possible to grow cactus and papyrus into a protected area

### ISSUE THREAD

**Issue by Montandalar** [minetest/minetest_game#2995]
Protection checks are in place in MTG to check that a tree when fully grown would not overlap a protected area. Therefore I think this check should apply to cactus and papyrus, which can grow 2 and 3 extra nodes taller after being planted respectively. This means if someone leaves a hole in the floor of their protected area, you can plant papyrus or cactus under their floor and its top 1-2 nodes will protrude up out of the floor:

![screenshot_20221025_222142](https://user-images.githubusercontent.com/13659694/197760172-b9414496-2354-4a4e-88d1-25442d8925d5.png)
![screenshot_20221025_223542](https://user-images.githubusercontent.com/13659694/197763504-114a510e-8b5f-4ee8-b0f4-b9b3c669a4b8.png)

In this instance the cobblestone is the first layer of the protected area.

Yes, this is kind of an edge case. Also, you can break papyrus by hand. You can't break cactus by hand though, so hypothetically someone could end up trapped.

**Comment by SmallJoker** [comment]
I don't think this needs fixing. Grass and flowers spread too, and cutting them is fairly trivial. Make sure that you've protected your lawn well enough.

**Comment by Montandalar** [comment]
I'm okay with that as an outcome, it would just mean we can solidify a "working as intended" response if anyone brings it up in future and link them here. Equally happy to go and fix it though depending on what others think.

**Comment by appgurueu** [comment]
> You can't break cactus by hand though, so hypothetically someone could end up trapped.

This is what worries me (as well as the inconsistency vs. trees).
